apps/src/studio/Sprite.js

```diff
@@ -0,0 +1,459 @@
+var Collidable = require('./collidable');
+var constants = require('./constants');
+var studioMsg = require('./locale');
+var spriteActions = require('./spriteActions');
+var Direction = constants.Direction;
+var Emotions = constants.Emotions;
+var NextTurn = constants.NextTurn;
+var utils = require('../utils');
+var _ = utils.getLodash();
+var StudioAnimation = require('./StudioAnimation');
+var StudioSpriteSheet = require('./StudioSpriteSheet');
+
+/**
+ * A Sprite is a type of Collidable.
+ * Note: x/y represent x/y of center in gridspace
+ * @extends {Collidable}
+ */
+var Sprite = function (options) {
+  // call collidable constructor
+  Collidable.apply(this, arguments);
+
+  /**
+   * Rendering offset for item animation vs display position - applied as
+   * late as possible.
+   * @type {{x: number, y: number}}
+   */
+  this.renderOffset = options.renderOffset || { x: 0, y: 0 };
+
+  this.speed = options.speed || constants.DEFAULT_SPRITE_SPEED;
+  this.displayDir = Direction.NONE;
+  this.startFadeTime = null;
+  this.fadeTime = 0;
+
+  this.image = null;
+  this.legacyImage = null;
+
+  this.animation_ = null;
+  this.legacyAnimation_ = null;
+
+  this.useLegacyIdleEmotionAnimations = false;
+
+  this.lastDrawPosition = { x: 0, y: 0 };
+};
+Sprite.inherits(Collidable);
+module.exports = Sprite;
+
+/**
+ * frameCounts objects are used in setImage and setLegacyImage:
```

Comment:
Optional documentation stuff: Great candidate for a [JSDoc typedef](http://usejsdoc.org/tags-typedef.html):

```javascript
/**
 * FrameCounts objects are used in setImage and setLegacyImage:
 * @typedef {Object} FrameCounts
 * @property {number} walkFrames - frames in walk animations (spriteSheet)
 * @property {number} idleFrames - frames in idle animations (spriteSheet)
 * @property {number} walk - default frames in all types of animations (spriteSheet)
 * etc...
 */
```

Then you can use the documented type in other documentation, e.g. for setImage:

```javascript
/**
 * Sets (or modifies) the image - we will generate a StudioSpriteSheet and StudioAnimation in response..
 * @param {string} image
 * @param {FrameCounts} frameCounts
 */
```